build.sh

```diff
@@ -1,385 +1,27 @@
-        arm64)
-            __BuildArch=arm64
-            ;;
-        debug)
-            __BuildType=Debug
-            ;;
-        release)
-            __BuildType=Release
-            ;;
-        clean)
-            __CleanBuild=1
-            ;;
-        verbose)
-            __VerboseBuild=1
-            ;;
-        clang3.5)
-            __ClangMajorVersion=3
-            __ClangMinorVersion=5
-            ;;
-        clang3.6)
-            __ClangMajorVersion=3
-            __ClangMinorVersion=6
-            ;;
-        clang3.7)
-            __ClangMajorVersion=3
-            __ClangMinorVersion=7
-            ;;
-        cross)
-            __CrossBuild=1
-            ;;
-        -dotnetclipath) 
-            shift
-            __dotnetclipath=$1
-        ;;
-        *)
-          __UnprocessedBuildArgs="$__UnprocessedBuildArgs $1"
-    esac
-    shift
-done
-
-# If neither managed nor native are passed as arguments, default to building both
-
-if [ "$__buildmanaged" = false -a "$__buildnative" = false ]; then
-    __buildmanaged=true
-    __buildnative=true
-fi
-
-# Set the remaining variables based upon the determined build configuration
-__IntermediatesDir="$__rootbinpath/obj/Native/$__BuildOS.$__BuildArch.$__BuildType"
-__ProductBinDir="$__rootbinpath/Product/$__BuildOS.$__BuildArch.$__BuildType"
-__RelativeProductBinDir="bin/Product/$__BuildOS.$__BuildArch.$__BuildType"
-
-# CI_SPECIFIC - On CI machines, $HOME may not be set. In such a case, create a subfolder and set the variable to set.
-# This is needed by CLI to function.
-if [ -z "$HOME" ]; then
-    if [ ! -d "$__ProjectRoot/temp_home" ]; then
-        mkdir "$__ProjectRoot/temp_home"
-    fi
-    export HOME=$__ProjectRoot/temp_home
-    echo "HOME not defined; setting it to $HOME"
-fi
-
-# Configure environment if we are doing a clean build.
-if [ $__CleanBuild == 1 ]; then
-    clean
-fi
-
-# Configure environment if we are doing a cross compile.
-if [ $__CrossBuild == 1 ]; then
-    export CROSSCOMPILE=1
-    if ! [[ -n "$ROOTFS_DIR" ]]; then
-        export ROOTFS_DIR="$__ProjectRoot/cross/rootfs/$__BuildArch"
-    fi
+# If setting variables failed, exit with the status code of the vars script
+if [ $BUILDERRORLEVEL != 0 ]; then
+    exit $BUILDERRORLEVEL
 fi
 
-setup_dirs
-
-if $__buildnative; then
-
-    # Check prereqs.
-
-    check_native_prereqs
-
-    # Prepare the system
-
-    prepare_native_build
-
-    # Build the corert native components.
+. ./cibuild/build-native.sh $*
 
-    build_native_corert
-
-    # Build complete
-fi
-
-# If native build failed, exit with the status code of the managed build
+# If native build failed, exit with the status code of the native build
 if [ $BUILDERRORLEVEL != 0 ]; then
     exit $BUILDERRORLEVEL
 fi
 
-if $__buildmanaged; then
-
-    # Prepare the system
-
-    prepare_managed_build
-
-    # Build the corert native components.
-
-    build_managed_corert
-
-    # Build complete
-fi
+. ./cibuild/build-managed.sh $*
```

Comment:
One of the comments on the initial PR was that we were dumping a lot of stuff in the root.  I can change the name to something more accurate like "buildscripts" or just put it in the root if we don't care.